pkgs/applications/video/screenkey/default.nix

```diff
@@ -0,0 +1,50 @@
+{ lib
+, substituteAll
+, buildPythonApplication
+, fetchFromGitHub
+, distutils_extra
+, setuptools-git
+, intltool
+, pygtk
+, libX11
+, libXtst
+}:
+buildPythonApplication rec {
+  pname = "screenkey";
+  version = "0.9";
+
+  src = fetchFromGitHub {
+    owner = "wavexx";
+    repo = "screenkey";
+    rev = "screenkey-${version}";
+    sha256 = "14g7fiv9n7m03djwz1pp5034pffi87ssvss9bc1q8vq0ksn23vrw";
+  };
+
+  patches = [
+    (substituteAll {
+      src = ./paths.patch;
+      inherit libX11 libXtst;
+    })
+  ];
+
+  nativeBuildInputs = [
+    distutils_extra
+    setuptools-git
+    intltool
```

Comment:
Adding `wrapGAppsHook` in `nativeBuildInputs` and `defaultIconTheme`, `hicolor_icon_theme` in `buildInputs` should provide icons regardless of the desktop environments. The other pull request for screenkey seems to do something similar: https://github.com/NixOS/nixpkgs/pull/34898